include/seqan3/alignment/pairwise/align_result.hpp

```diff
@@ -34,210 +34,157 @@
-    using base_type = detail::transfer_template_args_onto_t<output_type_list_t, std::tuple>;
+//! \brief Type deduction for id, score and end coordinate.
+template <typename id_t, typename score_t, typename coordinate_t>
+align_result_value_type(id_t, score_t, coordinate_t) -> align_result_value_type<id_t, score_t, coordinate_t>;
 
-    //!\brief Inheriting the constructors of the base type.
-    using base_type::base_type;
+//! \brief Type deduction without the trace.
+template <typename id_t, typename score_t, typename coordinate_t>
+align_result_value_type(id_t, score_t, coordinate_t, coordinate_t) -> align_result_value_type<id_t, score_t,
+                                                                                              coordinate_t>;
 
-    /*!\name Access functions
-     * \brief Convenience helper functions to access elements of the alignment result type.
-     * \{
-     */
+//! \brief Type deduction with all available fields.
+template <typename id_t, typename score_t, typename coordinate_t, typename trace_t>
+align_result_value_type(id_t, score_t, coordinate_t, coordinate_t, trace_t) -> align_result_value_type<id_t,
+                                                                                                       score_t,
+                                                                                                       coordinate_t,
+                                                                                                       trace_t>;
 
-    //!\brief Returns the id.
-    constexpr std::tuple_element_t<0, base_type> id() const noexcept
-    {
-        return std::get<0>(*this);
-    }
+} // namespace detail
 
-    //!\brief Returns the score.
-    constexpr std::tuple_element_t<1, base_type> score() const noexcept
-    {
-        return std::get<1>(*this);
-    }
+/*!\brief Stores the alignment results and gives access to score, traceback and the begin and end coordinates.
+ * \ingroup pairwise
+ * \tparam align_result_traits The type of the traits object.
+ *
+ * \details
+ * Objects of this class are the result of an alignment computation.
+ * It always contains and alignment identifier and the resulting score.
+ * Optionally – if the user requests – also the begin and end positions within
+ * the sequences and the trace can be calculated. When accessing a field that
+ * has not been calculated an std::bad_optional_access exception is thrown.
+ */
+template <typename align_result_traits>
+class align_result
+{
+private:
+    //! \brief Traits object that contains the actual alignment result data.
+    align_result_traits data;
+
+    using id_t = decltype(data.id);
+    using score_t = decltype(data.score);
+    using coord_t = decltype(data.end_coordinate);
+    using trace_t = decltype(data.trace);
+
+public:
+    /*!\brief Constructor to pass the alignment result traits.
+     * \param[in] traits The alignment results.
+     */
+    align_result(align_result_traits traits) : data(traits) {};
```

Comment:
value instead of traits? traits are used to augment the type.

Reply:
okay!